assets/activation-scripts/etc/profile.d/0800_cuda.sh

```diff
@@ -2,53 +2,51 @@
 export _coreutils="@coreutils@"
 export _gnused="@gnused@"
 export _findutils="@findutils@"
+export _ldconfig="@ldconfig@"
 # ============================================================================ #
 #
 # Setup CUDA
 #
 # ---------------------------------------------------------------------------- #
 
-# Only run if _FLOX_ENV_CUDA_DETECTION is set
 activate_cuda(){
   # Strip a trailing or lone slash so that we can construct it later.
   local fhs_root_prefix="${1%/}"
+  # Path to ldconfig that can be substituted for testing.
+  local ldconfig_bin="$2"
 
+  # Only run if _FLOX_ENV_CUDA_DETECTION is set
   if [[ "${_FLOX_ENV_CUDA_DETECTION:-}" != 1 ]]; then
     return 0
   fi
 
+  # Skip when not on Linux
+  if [[ "$ldconfig_bin" == "__LINUX_ONLY__" || ! -f "$ldconfig_bin" || ! -x "$ldconfig_bin" ]]; then
+    return 0
+  fi
+
+  # Skip when no Nvidia device
   if ! ( "$_findutils/bin/find" "${fhs_root_prefix}/dev" -maxdepth 1 -iname 'nvidia*' -o -iname dxg | read -r ;); then
     return 0
   fi
 
-  LIB_DIR="$("$_coreutils/bin/realpath" --no-symlinks "$FLOX_ENV/../../lib")"
-  SYSTEM_LIB_DIR=$("$_findutils/bin/find" \
-    "${fhs_root_prefix}/run/opengl-drivers" \
```

Comment:
I can't completely remember, but I think this path is for NixOS, and I'm not sure NixOS will have ldconfig. We probably need to keep that one. We should probably have been a bit more explicit about documenting these

Reply:
It seems so: https://alternativebit.fr/posts/nixos/nix-opengl-and-ubuntu-integration-nightmare/

It would be great if we didn't have to support two different implementations of this, one which uses `ldconfig` and another uses `find`, both of which need to pattern match the files and then join the results back together afterwards.

It looks like it's still possible to build a cache on a NixOS system. I can't verify whether it would definitely pick up entries in `/run/opengl-drivers` though:
```
dan@build-arm.floxdev.com ~ $ which ldconfig
/run/current-system/sw/bin/ldconfig
dan@build-arm.floxdev.com ~ $ ldconfig -XC ld.so.cache            
dan@build-arm.floxdev.com ~ $ ldconfig -pC ld.so.cache | head -n1
34 libs found in cache `ld.so.cache'
```

Would it be reasonable to ask people to generate their own cache in the known location `/etc/ld.so.cache`?

Reply:
I think the whole point of this feature is that users _don't_ need to do anything, so I'd prefer we had a fallback for when `ldconfig` doesnt' work. I don't have a ton of context on how this works, but if the logic is:
- Use `ldconfig` to find files
- If files are found, return them
- If files aren't found, search via `find` or `fd` (please use `fd`, it's much faster)
- If files are found, return them

That doesn't seem too bad if you factor out the "find files" functions.

Reply:
Turned out not to be so bad: 4a19d64145729b658f8c816b5466032c30c5b82c

Reply:
Except it didn't work on Linux. Investigating..

Reply:
Fixed in https://github.com/flox/flox/pull/1902/commits/b8c5cf75112340adb594cb87dfa4cdc78ba915a9 which was caused by:

- https://github.com/flox/flox/issues/1341

Can anyone verify whether the path should be `/run/opengl-drivers` (what we previously and currently have) or `/run/opengl-driver`? The article I linked above switches between both. GitHub code search has:

- [14 results for `/run/opengl-drivers` including ourselves](https://github.com/search?q=%22%2Frun%2Fopengl-drivers%22&type=code)
- [2.4k results for `/run/opengl-driver`](https://github.com/search?q=%22%2Frun%2Fopengl-driver%22&type=code)

cc @tomberek when you review this.

Reply:
It should be `/run/opengl-driver`

Reply:
Suggestions applied and PR description updated, thanks!